File: task_equivalents/llm_translations/llama/base/r/statistical_modelling_analysis/01_linear_regression_inference.R
# Load necessary libraries
library(readr)
library(broom)

# Read data from csv file
Auto <- read_csv('data/csv/Auto.csv', na = c("?", "NA")) %>% 
  drop_na()

# Prepare data for modeling
X <- Auto$horsepower
Y <- Auto$mpg

# Fit linear model
model1_fit <- lm(Y ~ X)

# Print model summary
summary(model1_fit)

# Make prediction for new data point
xnew <- data.frame(X = 98)
pred <- predict(model1_fit, newdata = xnew, interval = "confidence", level = 0.95)
pred_obs <- predict(model1_fit, newdata = xnew, interval = "prediction", level = 0.95)

# Print predictions
print(data.frame(mean = pred[, 1], mean_ci_lower = pred[, 2], mean_ci_upper = pred[, 3]))
print(data.frame(mean = pred[, 1], obs_ci_lower = pred_obs[, 2], obs_ci_upper = pred_obs[, 3]))

File: task_equivalents/llm_translations/llama/base/python/statistical_modelling_analysis/01_linear_regression_inference.py
import pandas as pd
import statsmodels.api as sm

# Load data
auto = pd.read_csv('data/csv/Auto.csv', na_values='?').dropna()

# Fit linear model
X = sm.add_constant(auto['horsepower'])
model1 = sm.OLS(auto['mpg'], X).fit()
print(model1.summary())

# Make predictions
new_data = pd.DataFrame({'const': 1, 'horsepower': [98]})
predict_conf = model1.get_prediction(new_data).summary_frame(alpha=0.05)[['mean', 'mean_ci_lower', 'mean_ci_upper']]
predict_pred = model1.get_prediction(new_data).summary_frame(alpha=0.05)[['mean', 'obs_ci_lower', 'obs_ci_upper']]

print(predict_conf)
print(predict_pred)